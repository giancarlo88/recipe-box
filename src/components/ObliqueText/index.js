import React from 'react'
import styled from 'styled-components'

const StyledObliqueText = styled.p`
font-style: oblique;
font-size: 12px;
padding: 7.5px 0;
`

const ObliqueText = ({text}) => (
  <StyledObliqueText>
    {text}
  </StyledObliqueText>
)

export default ObliqueText
