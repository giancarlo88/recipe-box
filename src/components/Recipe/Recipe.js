import React from 'react'
import styled from 'styled-components'

const Recipe = styled.div`
  background-color: #247BA0;
`

export default () => <Recipe>Here a recipe will go.</Recipe>
