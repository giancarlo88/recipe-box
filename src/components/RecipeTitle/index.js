import React from 'react'
import styled from 'styled-components'

const StyledRecipeTitle = styled.h2`
  font-weight: bold;
  font-size: 1.5em;
  padding: 15px 0;
`

const RecipeTitle = ({ text }) => <StyledRecipeTitle>{text}</StyledRecipeTitle>

export default RecipeTitle
