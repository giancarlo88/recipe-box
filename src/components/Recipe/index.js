import React from 'react'
import styled from 'styled-components'

import ObliqueText from '../ObliqueText'
import RecipeTitle from '../RecipeTitle'

const StyledRecipe = styled.div`
  background-color: #247ba0;
  color: #C9DAEA;
  height: ${(props) => (props.isSelected ? '400px' : '200px')};
  padding: 15px;
  transition: height .2s ease-in;
`

const Recipe = ({ handleClick, isSelected, recipe, index}) => (
  <StyledRecipe isSelected={isSelected} onClick={() => handleClick(index)}>
    <RecipeTitle text={recipe.title} />
    <ObliqueText text={`Serves ${recipe.servings}`} />
    <ul>
      {recipe.ingredients.map((ingredient, ingredientIndex) => (
        <li key={ingredientIndex}>
          {ingredient.quantity} {ingredient.unit} {ingredient.ingredient}
        </li>
      ))}
    </ul>
    <br />
    <p>{recipe.description}</p>
  </StyledRecipe>
)

Recipe.defaultProps = {
  recipe: {
    title: '',
    servings: 0,
    ingredients: [],
    description: ''
  }
}

export default Recipe
