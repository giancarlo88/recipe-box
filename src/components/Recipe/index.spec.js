import Recipe from './'
import React from 'react'
import { create } from 'react-test-renderer'
import { mount } from 'enzyme'

import RecipeTitle from '../RecipeTitle'
import ObliqueText from '../ObliqueText'



describe('<Recipe />', () => {
  const title = 'Test recipe'
  const servings = 3
  const ingredients = [
    {
      quantity: 1,
      unit: 'cup',
      ingredient: 'flour'
    }
  ]
  const description = 'Here is my test recipe'

  const recipe = {
    title,
    servings,
    ingredients,
    description
  }
  it('should match snapshot', () => {
    const props = {
      recipe
    }
    const tree = create(<Recipe {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('should render the recipe passed to it from props', () => {
    const component = mount(<Recipe recipe={recipe} />)
    expect(component.find(RecipeTitle).props().text).toBe(title)
    expect(component.find(ObliqueText).props().text).toBe(`Serves ${servings}`)
    expect(component.find('li').text()).toBe(`${ingredients[0].quantity} ${ingredients[0].unit} ${ingredients[0].ingredient}`)
    expect(component.find('p').at(1).text()).toBe(description)
  })
})
