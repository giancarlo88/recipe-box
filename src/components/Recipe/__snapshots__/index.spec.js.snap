// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipe /> should match snapshot 1`] = `
<div
  className="sc-htpNat jdaKho"
  onClick={[Function]}
>
  <h2
    className="sc-bwzfXH cIKGFn"
  >
    Test recipe
  </h2>
  <p
    className="sc-bdVaJa hCJUJ"
  >
    Serves 3
  </p>
  <ul>
    <li>
      1
       
      cup
       
      flour
    </li>
  </ul>
  <br />
  <p>
    Here is my test recipe
  </p>
</div>
`;
